### PR TITLE
1.6 GA Docs changes

### PR DESCRIPTION
**Pull request by matthewshaver**
## What are you changing in this pull request and why?

Changes to the docs pages to take 1.6 GA - adjusting prerelease tags, changing migration guide titles, and adjusting release table. 

## ChecklistUncomment if you're publishing docs for a prerelease version of dbt (delete if not applicable):
- [x] Add versioning components, as described in [Versioning Docs](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-entire-pages)
- [x] Add a note to the prerelease version [Migration Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/guides/migration/versions)

- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [x] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

### CODE REVIEW

File: website/docs/guides/migration/versions/01-upgrading-to-v1.6.md [hunk 1]
```diff
@@ -1,16 +1,8 @@
...
 
-:::warning Prerelease
-
-dbt Core v1.6 is available as a release candidate. [Final release is planned for July 31.](https://github.com/dbt-labs/dbt-core/issues/7990)
-
-Test it out, and [let us know](https://github.com/dbt-labs/dbt-core/issues/new/choose) if you run into any issues!
-
-:::
-
 ## Resources
```
Comment (runleonarun) [inline website/docs/guides/migration/versions/01-upgrading-to-v1.6.md:7]: @matthewshaver I see a TODO a little lower on this page:  `[TODO] changes`

File: website/docs/guides/migration/versions/01-upgrading-to-v1.6.md [hunk 2]
```diff
@@ -1,16 +1,8 @@
...
 
 - [Changelog](https://github.com/dbt-labs/dbt-core/blob/1.6.latest/CHANGELOG.md)
```
Comment (runleonarun) [inline website/docs/guides/migration/versions/01-upgrading-to-v1.6.md:8]: @matthewshaver I see a TODO a little lower on this page: 
> `[TODO] changes`

File: website/snippets/core-versions-table.md [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
+| [**v1.4**](/guides/migration/versions/upgrading-to-v1.4)   | Jan 25, 2023    | Critical       | Jan 25, 2024            | 
+| [**v1.3**](/guides/migration/versions/upgrading-to-v1.3)   | Oct 12, 2022    | Critical       | Oct 12, 2023            | 
+| [**v1.2**](/guides/migration/versions/upgrading-to-v1.2)   | Jul 26, 2022    | End of Life* ⚠️ | Jul 26, 2023            |
 | [**v1.1**](/guides/migration/versions/upgrading-to-v1.1) ⚠️ | Apr 28, 2022    | End of Life* ⚠️ | Apr 28, 2023            | 
 | [**v1.0**](/guides/migration/versions/upgrading-to-v1.0) ⚠️ | Dec 3, 2021     | End of Life* ⚠️ | Dec 3, 2022 ⚠️           | 
-|  **v0.X** ⛔️                                                | (Various dates) | Deprecated ⛔️ | Deprecated ⛔️            | 
+|  **v0.X** ⛔️                                               | (Various dates) | Deprecated ⛔️  | Deprecated ⛔️            | 
 _*All versions of dbt Core since v1.0 are available in dbt Cloud until further notice. Versions that are EOL do not receive any fixes. For the best support, we recommend upgrading to a version released within the past 12 months._
 ### Planned future releases
```
Comment (runleonarun) [inline website/snippets/core-versions-table.md:15]: Should we remove 1.6 from the upcoming versions table below?

**Review by runleonarun — CHANGES_REQUESTED** [review]
Just two small comments, otherwise looks good!

File: website/docs/guides/migration/versions/01-upgrading-to-v1.6.md (hunk 2 above)
Comment (matthewshaver) [reply website/docs/guides/migration/versions/01-upgrading-to-v1.6.md:8]: Thank you! Thought I caught them all

File: website/snippets/core-versions-table.md (hunk 3 above)
Comment (matthewshaver) [reply website/snippets/core-versions-table.md:15]: Good catch! Thank you

**Review by runleonarun — APPROVED** [review]
Looks ready to 🚢

**Pull request merged**